src/cli/common.rs

```diff
@@ -375,88 +376,55 @@ where
     Ok(utils::ExitCode(0))
 }
 
-pub(crate) fn list_targets(distributable: DistributableToolchain<'_>) -> Result<utils::ExitCode> {
-    let mut t = process().stdout().terminal();
-    let manifestation = distributable.get_manifestation()?;
-    let config = manifestation.read_config()?.unwrap_or_default();
-    let manifest = distributable.get_manifest()?;
-    let components = manifest.query_components(distributable.desc(), &config)?;
-    for component in components {
-        if component.component.short_name_in_manifest() == "rust-std" {
-            let target = component
-                .component
-                .target
-                .as_ref()
-                .expect("rust-std should have a target");
-            if component.installed {
-                let _ = t.attr(terminalsource::Attr::Bold);
-                let _ = writeln!(t.lock(), "{target} (installed)");
-                let _ = t.reset();
-            } else if component.available {
-                let _ = writeln!(t.lock(), "{target}");
-            }
-        }
-    }
-
-    Ok(utils::ExitCode(0))
+pub(crate) fn list_targets(
+    distributable: DistributableToolchain<'_>,
+    installed_only: bool,
```

Comment:
I don't know what you think in terms of API design, but I remember having read somewhere that Rust "hates boolean parameters", since you basically don't know what boolean literals mean in a function call.

Your particular uses of this function look perfectly fine, but considering possible future uses (esp. from external contributors), I would've refactored this into two `pub(crate) fn`s such as `list_all_targets()` and `list_installed_targets()` calling one private implementation that has that boolean parameter...

I'll let you decide on this one.